File: src/main/java/com/socialhobbies/shuserservice/entity/Account.java
package com.socialhobbies.shuserservice.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
@Table(name = "account")
public class Account {

    @Id
    @Column(name = "account_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonIgnore
    @OneToOne(optional = false, mappedBy = "account")
    private User user;

    @JoinColumn(nullable = false)
    private Long amount = 0L;

    public Account() {
    }

    public Account(User user) {
        this.user = user;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Long getAmount() {
        return amount;
    }

    public void setAmount(Long amount) {
        this.amount = amount;
    }

    @Override
    public String toString() {
        return "Account{" +
                "id=" + id +
                ", user=" + user +
                ", amount=" + amount +
                '}';
    }
}

File: src/main/java/com/smarthome/shuserservice/entity/Account.kt
package com.smarthome.shuserservice.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
class Account(
    @Id
    @Column(name = "account_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @JsonIgnore
    @OneToOne(optional = false, mappedBy = "profile")
    val userOld: UserOld? = null,

    var amount: Long = 0L,
)
